### PR TITLE
yaml callback plugin: deprecation

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

The plugin itself has a note stating that since ansible-core 2.13 (as of Dec '24 the minimum verison of ansible-core supported by community.general) has the functionality implemented within its default callback plugin.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/callback/yaml.py

### CODE REVIEW

File: plugins/callback/yaml.py [hunk 1]
```diff
@@ -12,6 +12,10 @@
     name: yaml
     type: stdout
     short_description: YAML-ized Ansible screen output
+    deprecated:
+        removed_in: 12.0.0
+        why: Starting in ansible-core 2.13, the default callback has support for printing output in YAML format.
```
Comment (felixfontein) [inline plugins/callback/yaml.py:17]:
```suggestion
        why: Starting in ansible-core 2.13, the P(ansible.builtin.default#callback) callback has support for printing output in YAML format.
```

**Review by felixfontein — COMMENTED** [review]
Since yaml is (or at least used to be, not sure how many migrated to `result_format=yaml`) one of the most used callback plugins, it might be warranted to have a longer deprecation period. What do you think of only removing it from 13.0.0, so users have a bit more than a full year to adjust?

**Comment by russoz** [comment]
> Since yaml is (or at least used to be, not sure how many migrated to `result_format=yaml`) one of the most used callback plugins, it might be warranted to have a longer deprecation period. What do you think of only removing it from 13.0.0, so users have a bit more than a full year to adjust?

agreed!

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz it looks like this now has conflicts as well...

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for your contribution!

**Pull request merged**